tests/pass/issues/issue-miri-3680.rs

```diff
@@ -0,0 +1,24 @@
+//@ignore-target-windows: File handling is not implemented yet
+//@compile-flags: -Zmiri-disable-isolation
+
+use std::fs::remove_file;
+use std::io::Seek;
+use std::panic::catch_unwind;
+
+#[path = "../../utils/mod.rs"]
+mod utils;
+
+fn main() {
+    let path = utils::prepare("miri_test_fs_seek_i64_max_plus_1.txt");
+
+    let result = catch_unwind(|| {
+        let mut f = std::fs::File::create(&path).unwrap();
+        f.seek(std::io::SeekFrom::Start(i64::MAX as u64 + 1)).unwrap();
```

Comment:
Looks like on a Windows host, we are getting "unsupported io error". Curious that this doesn't happen on Linux hosts, I guess they generate different `io::ErrorKind` for this case?

Reply:
There's no named constant for error 131, so on windows this becomes `ErrorKind::Other` and then we can't map it to a target error.

I think if the offset is negative we probably don't want to call `file_descriptor.seek` at all, and instead return EINVAL directly ourselves.

Reply:
Indeed. So I guess test should only be run for unix host. No idea how to configure this, but `@only-host-linux` is good enough?

Sounds good.

Reply:
No it should run on all hosts, Miri should just behave consistently.

Reply:
Oh, I got it. We only need to know Miri can interpret this operation successfully without ICE so the stderr output doesn't matter.
How can I configure the test header for this?  I tried `@rustc-env: MIRI_SKIP_UI_CHECKS=1` and it still compares output.

Reply:
No stderr output should be checked.

If you do this...

> I think if the offset is negative we probably don't want to call file_descriptor.seek at all, and instead return EINVAL directly ourselves.

... then I think all the troubles with the test will disappear.

Reply:
Understood now. No need to panic on purpose thus no stderr is emitted.

Reply:
No, that's not what I meant. The error code will be the same across all targets if you  implement what I quoted above, so the stderr will be the same.